File: server/Models/Guide.cs
namespace AFCSite.Server.Models;

public class Guide
{
    public required string Slug { get; set; }
    public required string Title { get; set; }
    public required string Category { get; set; }
    public required string ReadTime { get; set; }
    public required string Image { get; set; }
    public required string Intro { get; set; }
    public required List<GuideSection> Sections { get; set; }
}

public class GuideSection
{
    public required string Heading { get; set; }
    public required string Content { get; set; }
}

File: src/app/models/guide.ts
export interface GuideSection {
  heading: string;
  content: string;
}

export interface Guide {
  slug: string;
  title: string;
  category: string;
  readTime: string;
  image: string;
  intro: string;
  sections: GuideSection[];
}

export type GuideSummary = Pick<Guide, 'slug' | 'title' | 'category' | 'readTime' | 'image' | 'intro'>;
